internal/cache/fifo.go

```diff
@@ -55,4 +63,5 @@ func (f *FIFO[K, V]) Get(key K) (V, bool) {
 // [WriteLock] is held when this is accessed.
 func (f *FIFO[K, V]) remove(key K) {
 	delete(f.m, key)
+	f.removedKey = &key
```

Comment:
Hmm it seems fairly odd to pass around this data via an internal pointer modified via the callback in the bounded buffer.

It seems that we added this to return the removed key for the indexer, but I don't think we should modify the fifo cache with this hack to support that specific case. Is there a reason not to use `DeleteRange` to make sure that we've removed everything below our desired height?